packages/Modal/Modal.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { useRef, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import Box from '@tds/core-box'
+import Button from '@tds/core-button'
+import Paragraph from '@tds/core-paragraph'
+import { safeRest } from '@tds/util-helpers'
+import { Close } from '@tds/core-interactive-icon'
+import Heading from '@tds/core-heading'
+import {
+  StyledModal,
+  CTAWrapper,
+  CloseButtonWrapper,
+  FullScreenOverlay,
+  ModalWrapper,
+  PaddingOverride,
+  OutlineButton,
+} from './styles'
+
+const Modal = ({
+  heading,
+  confirmCTAText,
+  cancelCTAText,
+  bodyText,
+  modalOpen,
+  closeModalHandler,
+  proceedModalHandler,
+  focusElementAfterClose,
+  ...rest
+}) => {
+  const ModalOverlayRef = useRef(null)
+  const modalRef = useRef(null)
+  const header = useRef(null)
+  const firstCTA = useRef(null)
+  const closeButton = useRef(null)
+
+  const tabbingHandling = e => {
+    if (document.activeElement === firstCTA.current) {
+      e.preventDefault()
+      return closeButton.current.focus()
+    }
+    if (document.activeElement === closeButton.current) {
+      e.preventDefault()
+      return firstCTA.current.focus()
+    }
+    return null
+  }
+
+  const closingModal = () => {
+    if (focusElementAfterClose && focusElementAfterClose.current) {
+      focusElementAfterClose.current.focus()
+    }
+    return closeModalHandler()
+  }
+
+  const handleKeyDown = e => {
+    if (e.key === 'Escape') {
+      return closingModal()
+    }
+    if ((e.shiftKey && e.key === 'Tab') || e.key === 'Tab') {
+      return tabbingHandling(e)
+    }
+    return null
+  }
+
+  const preventScroll = e => {
+    if (ModalOverlayRef.current.contains(e.targetTouches[0].target)) {
+      e.preventDefault()
+    }
+  }
+
+  const removeEventScrolling = () => {
+    document.body.removeEventListener('touchmove', preventScroll)
+    document.body.style.overflow = 'visible'
+  }
+
+  const handleOutSideClick = e => {
+    if (!modalRef.current.contains(e.target)) {
+      closingModal()
+    }
+  }
+
+  useEffect(() => {
+    if (modalOpen) {
+      document.body.addEventListener('touchmove', preventScroll, { passive: false })
+      document.body.addEventListener('keydown', handleKeyDown, { passive: false })
+      document.body.addEventListener('mousedown', handleOutSideClick, { passive: false })
+      document.body.style.overflow = 'hidden'
+      header.current.focus()
+    } else {
+      removeEventScrolling()
+      document.body.removeEventListener('keydown', handleKeyDown)
+    }
+    return () => {
+      removeEventScrolling()
+      document.body.removeEventListener('keydown', handleKeyDown)
+      document.body.removeEventListener('mousedown', handleOutSideClick)
+    }
+  })
+
+  return (
+    <React.Fragment>
```

Comment:
This Fragment is unnecessary since there's already a single parent wrapper.

Reply:
I believe it is necessary, what single parent wrapper are you referring to?
If I remove the fragment then the component would be wrapped with {modalOpen...} invalid syntax

Reply:
It looks like we need React.Fragment here due to the conditional modelOpen that is wrapping the component. @theetrain